client/src/index.jsx

```diff
@@ -14,13 +15,34 @@ const Wrapper = Styled.div`
 class App extends React.Component {
   constructor() {
     super();
+    
+    this.state = {
+      locations: [],
+      materials: []
+    }
+    this.getProductById = this.getProductById.bind(this);
   }
   
+  getProductById() {
+    let id = window.location.pathname;
+    axios.get(`/locations${id}`)
```

Comment:
I've heard its best practice to call this.setState only once if you can. This could be refactored with promise.all and destructuring, and then setState could be called once with both datas